src/VuData.cpp

```diff
@@ -1014,22 +997,22 @@ PVR_ERROR Vu::GetChannelGroupMembers(ADDON_HANDLE handle, const PVR_CHANNEL_GROU
  * Channels
  **************************************************************************/
 
-int Vu::GetChannelsAmount()
+int Vu::GetChannelsAmount() const
 {
   return m_channels.size();
 }
 
-int Vu::GetChannelNumber(std::string strServiceReference)  
+int Vu::GetChannelNumber(std::string strServiceReference) const
 {
-  for (unsigned int i = 0;i<m_channels.size();  i++) 
+  for (const auto& channel : m_channels)
   {
-    if (!strServiceReference.compare(m_channels[i].strServiceReference))
-      return i+1;
+    if (!strServiceReference.compare(channel.strServiceReference))
+      return channel.iChannelNumber;
```

Comment:
does this change behavior?

Reply:
Nope, iChannelNumber is only set here: newChannel.iChannelNumber = m_channels.size()+1; before it's emplaced on the vector.

We never delete channels so ChannelNumber is always equal to vector index+1;

Reply:
okay